os/arch/arm/src/amebasmart/amebasmart_mipi.c

```diff
@@ -0,0 +1,352 @@
+	case MIPI_VIDEO_DATA_FORMAT_RGB888:
+	default:
+		bit_per_pixel = 24;
+		break;
+	}
+	priv->MIPI_InitStruct->MIPI_LaneNum = 2;
+	priv->MIPI_InitStruct->MIPI_FrameRate = MIPI_FRAME_RATE;
+	priv->MIPI_InitStruct->MIPI_HSA = MIPI_DSI_HSA * bit_per_pixel / 8;	//- 10; /* here the unit is pixel but not us */
+	if (priv->MIPI_InitStruct->MIPI_VideoModeInterface == MIPI_VIDEO_NON_BURST_MODE_WITH_SYNC_PULSES) {
+		priv->MIPI_InitStruct->MIPI_HBP = MIPI_DSI_HBP * bit_per_pixel / 8;	//- 10;
+	} else {
+		priv->MIPI_InitStruct->MIPI_HBP = (MIPI_DSI_HSA + MIPI_DSI_HBP) * bit_per_pixel / 8;	//-10 ;
+	}
+	priv->MIPI_InitStruct->MIPI_HACT = dsi_host->config.XPixels;
+	priv->MIPI_InitStruct->MIPI_HFP = MIPI_DSI_HFP * bit_per_pixel / 8;	//-12;
+	priv->MIPI_InitStruct->MIPI_VSA = MIPI_DSI_VSA;
+	priv->MIPI_InitStruct->MIPI_VBP = MIPI_DSI_VBP;
+	priv->MIPI_InitStruct->MIPI_VACT = dsi_host->config.YPixels;
+	priv->MIPI_InitStruct->MIPI_VFP = MIPI_DSI_VFP;
+	vtotal = priv->MIPI_InitStruct->MIPI_VSA + priv->MIPI_InitStruct->MIPI_VBP + priv->MIPI_InitStruct->MIPI_VACT + priv->MIPI_InitStruct->MIPI_VFP;
+	htotal_bits = (MIPI_DSI_HSA + MIPI_DSI_HBP + priv->MIPI_InitStruct->MIPI_HACT + MIPI_DSI_HFP) * bit_per_pixel;
+	overhead_cycles = T_LPX + T_HS_PREP + T_HS_ZERO + T_HS_TRAIL + T_HS_EXIT;
+	overhead_bits = overhead_cycles * priv->MIPI_InitStruct->MIPI_LaneNum * 8;
+	total_bits = htotal_bits + overhead_bits;
+	priv->MIPI_InitStruct->MIPI_VideDataLaneFreq = priv->MIPI_InitStruct->MIPI_FrameRate * total_bits * vtotal / priv->MIPI_InitStruct->MIPI_LaneNum / Mhz + 20;
+	priv->MIPI_InitStruct->MIPI_LineTime = (priv->MIPI_InitStruct->MIPI_VideDataLaneFreq * Mhz) / 8 / priv->MIPI_InitStruct->MIPI_FrameRate / vtotal;
+	priv->MIPI_InitStruct->MIPI_BllpLen = priv->MIPI_InitStruct->MIPI_LineTime / 2;
+
+	if (MIPI_DSI_HSA + MIPI_DSI_HBP + dsi_host->config.XPixels + MIPI_DSI_HFP < (512 + MIPI_DSI_RTNI * 16)) {
+		dbg("!!ERROR!!, LCM NOT SUPPORT\n");
+	}
+	if (priv->MIPI_InitStruct->MIPI_LineTime * priv->MIPI_InitStruct->MIPI_LaneNum < total_bits / 8) {
+		dbg("!!ERROR!!, LINE TIME TOO SHORT!\n");
+	}
+	if (priv->MIPI_InitStruct->MIPI_VideDataLaneFreq * 2 / 24 >= vo_freq) {
+		dbg("!!ERROR!!, VO CLK too slow!\n");
+	}
+	dbg("DataLaneFreq: %d, LineTime: %d\n", priv->MIPI_InitStruct->MIPI_VideDataLaneFreq, priv->MIPI_InitStruct->MIPI_LineTime);
+}
+
+static void amebasmart_mipi_init_helper(FAR struct amebasmart_mipi_dsi_host_s *priv)
+{
+	amebasmart_check_freq(&(priv->dsi_host.config));
+	amebasmart_set_clock();
+
+	MIPI_StructInit(priv->MIPI_InitStruct);
+	amebasmart_mipi_fill_buffer(&(priv->dsi_host));
+	MIPI_Init(priv->MIPIx, priv->MIPI_InitStruct);
+
+	MIPI_DSI_TO1_Set(priv->MIPIx, DISABLE, 0);
+	MIPI_DSI_TO2_Set(priv->MIPIx, ENABLE, 0x7FFFFFFF);
+	MIPI_DSI_TO3_Set(priv->MIPIx, DISABLE, 0);
+
+	amebasmart_register_interrupt();
+	MIPI_DSI_INT_Config(priv->MIPIx, DISABLE, ENABLE, FALSE);
+	MIPI_DSI_init(priv->MIPIx, priv->MIPI_InitStruct);
```

Comment:
The `MIPI_DSI_init` here can be removed, it has been satisfied in `MIPI_Init `already.